### ISSUE TITLE
Py-BoostLSS: An extension of Py-Boost to probabilistic modelling

### ISSUE THREAD

**Issue by StatMixedML** [sb-ai-lab/Py-Boost#1]
Dear Py-Boost developers,

Thanks for the very interesting paper and for making the code publicly available. 

I am the author of [XGBoostLSS](https://github.com/StatMixedML/XGBoostLSS) and [LightGBMLSS](https://github.com/StatMixedML/LightGBMLSS) that extend the base implementations to a probabilistic setting, where all moments of a parametric univariate and multivariate distribution are modeled as functions of covariates. This allows one to create probabilistic predictions from which intervals and quantiles of interest can be derived.

However, as outlined in my latest paper [März, Alexander (2022), Multi-Target XGBoostLSS Regression](https://arxiv.org/abs/2210.06831), XGBoost does not scale very well for the multivariate setting, since a separate tree is grown for each parameter individually. As an example, consider modelling a multivariate Gaussian distribution with `D=100` target variables, where the covariance matrix is approximated using the Cholesky-Decomposition. Modelling all conditional moments (i.e., means, standard-deviations and all pairwise correlations) requires estimation of `D(D + 3)/2 = 5,150` parameters.

I came across your approach just recently and spent the last few days extending your base model to [Py-BoostLSS: An extension of Py-Boost to probabilistic modelling](https://github.com/StatMixedML/Py-BoostLSS). Because it is very runtime efficient, SketchBoost is a good candidate for estimating high-dimensional target variables. The package is in a very early stage and I need to evaluate the runtime efficiency against XGBoostLSS.

@btbpanda I was wondering if you would be interested in a scientific collaboration to further extend the functionality of Py-BoostLSS. Looking forward to your reply.

**Comment by btbpanda** [comment]
Hey, @StatMixedML @valeman !

@StatMixedML Thanks for your feedbak. I am happy you like our work and find it useful to your research. I believe that what you propose is one of the features we shuold implement and I have probabilistic modelling in my TODO list. So we definetly should make the research collaboration. If you need any help or ideas how to implement some features inside the Py-Boost - just let me know. I am going to learn your repo and paper soon. 

There are however the limitation, that you need to keep in mind working with multioutput targets with Py-Boost. It is single GPU only, may be latter I will add Multi GPU, but now your dataset should fit the GPU memory. For 5k dimensions it could be problem if you have many rows. Before I tried on 2.5k targets at max for the real world datasets. But yes, if it fits, it will be order of magnitude faster than others.

@valeman thanks to your reply. I think, Py-Boost is flexible enough to implement multiple approaches so we should check in practice what performs better in terms of both speed and accuracy

**Comment by StatMixedML** [comment]
> @StatMixedML Thanks for your feedbak. I am happy you like our work and find it useful to your research. I believe that what you propose is one of the features we shuold implement and I have probabilistic modelling in my TODO list. So we definetly should make the research collaboration. If you need any help or ideas how to implement some features inside the Py-Boost - just let me know. I am going to learn your repo and paper soon.

Thanks for your feedback. Sound great. Let me know if you have some comments/suggestions on the repo. Please note, it is in a very early stage and subject to improvements. I suggest you start reading the Multi-Target XGBoostLSS paper first to get started.
